cdap-master/src/main/java/io/cdap/cdap/master/upgrade/UpgradeJobMain.java

```diff
@@ -110,6 +110,8 @@ private static void suspendSchedulesAndStopPipelines(ClientConfig clientConfig)
               scheduleClient.suspend(scheduleId);
             }
           }
+          // Need to stop workflows first or else the program will fail to stop below
+          programClient.stop(workflowId);
```

Comment:
This is a non-blocking stop.
Batch programs may take many seconds (a minute or so) to be stopped after this call. Streaming programs may take longer. 

After calling `stop`, would you want to [wait for them to be stopped](https://github.com/cdapio/cdap/blob/develop/cdap-client/src/main/java/io/cdap/cdap/client/ProgramClient.java#L278)?

Reply:
Ah, you call `stopAll` below, which blocks internally, so nevermind.